examples/fips/README.md

```diff
@@ -0,0 +1,83 @@
+# FIPS Compliance
+
+We are testing and making this client FIPS compliant using OpenSSL 3.0. Use OpenSSL 3.0 to use this client in FIPS compliant mode. Older versions of OpenSSL might work in similar way but we have not verified that.
+
+## Communication between client and Kafka cluster
+
+### Install from prebuilt wheels
+
+If this client is installed using `pip`, it comes with librdkafka in which OpenSSL 3.0 is statically linked. This client can communicates with Kafka cluster with FIPS approved algorithms using FIPS provider provided by OpenSSL. Steps to enable FIPS provider is mentioned in section [Enabling FIPS provider](#enabling-fips-provider).
+
+
+### Install from source by building librdkafka
+
+When building librdkafka and python client from source, librdkafka uses OpenSSL present in the system by dynamically linking it. If the installed OpenSSL is already working in FIPS mode, then you can directly jump to section [Client configuration to enable FIPS provider](#client-configuration-to-enable-fips-provider), otherwise use steps mentioned in section [Enabling FIPS provider](#enabling-fips-provider) to make system OpenSSL FIPS compliant. Once that is done, librdkafka and python client will use FIPS approved algorithms for the communication between client and Kafka Cluster using producer, consumer or admin client.
+
+### Enabling FIPS provider
+
+We assume that you have FIPS provider (module) available in your system. This module can be dynamically plugged into OpenSSL by putting this module in default module folder of OpenSSL or pointing to this module with environment vairable `OPENSSL_MODULES` (folder containing the FIPS module in the form of dynamic library). We also need to use OpenSSL config which has FIPS configuration present in it. Default configuration file can be modified to include FIPS related config or a new configuration file can also be pointed using `OPENSSL_CONF` environment variable.
```

Comment:
```suggestion
The FIPS provider is a module. To enable the FIPS provider, you must have the module available on your system, plug the module into OpenSSL, and then configure OpenSSL to use the module. 

You can plug the FIPS provider into OpenSSL two ways: 1) put the module in the default module folder of OpenSSL or 2) point to the module with the environment variable, `OPENSSL_MODULES`. For example: `OPENSSL_MODULES="/path/to/fips/module/lib/folder/` 

You configure OpenSSL to use the FIPS provider using the FIPS configuration in OpenSSL config. You can modify the default configuration file to include FIPS related config or create a new configuration file point to it using the environment variable,`OPENSSL_CONF`. For example `OPENSSL_CONF="/path/to/fips/enabled/openssl/config/openssl.cnf`: 
```
This might be improved by adding what needs to be configured in OpenSSL config.

Reply:
Used your suggestion with some changes. Please check.

Reply:
@pranavrth the changes above look good to me. I have looked and have no further recommendations.